python/tests/unit/processes/local_runtime/test_local_step.py

```diff
@@ -0,0 +1,558 @@
+from semantic_kernel.exceptions.process_exceptions import (
+    ProcessFunctionNotFoundException,
+)
+from semantic_kernel.functions import KernelFunction
+from semantic_kernel.functions.kernel_function_metadata import KernelFunctionMetadata
+from semantic_kernel.processes.kernel_process.kernel_process import KernelProcess
+from semantic_kernel.processes.kernel_process.kernel_process_edge import KernelProcessEdge
+from semantic_kernel.processes.kernel_process.kernel_process_event import KernelProcessEvent
+from semantic_kernel.processes.kernel_process.kernel_process_function_target import KernelProcessFunctionTarget
+from semantic_kernel.processes.kernel_process.kernel_process_state import KernelProcessState
+from semantic_kernel.processes.kernel_process.kernel_process_step_info import KernelProcessStepInfo
+from semantic_kernel.processes.kernel_process.kernel_process_step_state import KernelProcessStepState
+from semantic_kernel.processes.local_runtime.local_event import LocalEvent
+from semantic_kernel.processes.local_runtime.local_message import LocalMessage
+from semantic_kernel.processes.local_runtime.local_step import LocalStep
+
+
+async def test_parse_initial_configuration_adds_ids_if_missing():
+    """Test that parse_initial_configuration sets the step_info.state.id if it is None."""
+    data = {
+        "step_info": KernelProcess(
+            state=KernelProcessState(name="test_step"),  # noqa: F821
+            steps=[AsyncMock(spec=KernelProcessStepInfo)],
+        ),
+    }
+
+    # Call the parse_initial_configuration
+    processed_data = LocalStep.parse_initial_configuration(data)  # type: ignore
+
+    # Assert that it sets the id
+    assert processed_data["step_state"].id is not None
+    assert processed_data["event_namespace"] == f"{processed_data['step_state'].name}_{processed_data['step_state'].id}"
+    assert "output_edges" in processed_data
+
+
+def test_name_property():
+    """Test that the name property returns the name from the step_info.state."""
+    mock_state = KernelProcessStepState(name="my_step", id="123", state=None)
+    mock_info = KernelProcessStepInfo(inner_step_type=AsyncMock(spec=type), state=mock_state, output_edges={})
+
+    step = LocalStep(
+        kernel=MagicMock(spec=Kernel),
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+    )
+
+    assert step.name == "my_step"
+
+
+def test_id_property_with_id():
+    """Test that the id property returns the ID if it is available."""
+    mock_state = KernelProcessStepState(name="my_step", id="123", state=None)
+    mock_info = KernelProcessStepInfo(inner_step_type=AsyncMock(spec=type), state=mock_state, output_edges={})
+
+    step = LocalStep(
+        kernel=MagicMock(spec=Kernel),
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+    )
+
+    assert step.id == "123"
+
+
+def test_id_property_without_id():
```

Comment:
Can we combine with the previous test and `@pytest.mark.parametrize` to avoid duplication?